t/unit/worker/test_consumer.py

```diff
@@ -306,6 +322,63 @@ def test_cancel_long_running_tasks_on_connection_loss__warning(self):
         with pytest.deprecated_call(match=CANCEL_TASKS_BY_DEFAULT):
             c.on_connection_error_after_connected(Mock())
 
+    @pytest.mark.parametrize("broker_connection_retry", [True, False])
+    @pytest.mark.parametrize("broker_connection_retry_on_startup", [None, False])
+    def test_ensure_connected(self, subtests, broker_connection_retry, broker_connection_retry_on_startup):
+        c = self.get_consumer()
+        c.app.conf.broker_connection_retry_on_startup = broker_connection_retry_on_startup
+        c.app.conf.broker_connection_retry = broker_connection_retry
+
+        if broker_connection_retry_on_startup is None:
+            with subtests.test("Deprecation warning when startup is None"):
+                with pytest.deprecated_call():
+                    c.ensure_connected(Mock())
+
+        if broker_connection_retry is False:
+            with subtests.test("Does not retry when connect throws an error and retry is set to false"):
+                conn = Mock()
+                conn.connect.side_effect = ConnectionError()
+                with pytest.raises(ConnectionError):
+                    c.ensure_connected(conn)
+
+
+@pytest.mark.parametrize(
+    "broker_connection_retry_on_startup,is_connection_loss_on_startup",
+    [
+        pytest.param(False, True, id='shutdown on connection loss on startup'),
+        pytest.param(None, True, id='shutdown on connection loss on startup when retry on startup is undefined'),
+        pytest.param(False, False, id='shutdown on connection loss not on startup but startup is defined as false'),
+        pytest.param(None, False, id='shutdown on connection loss not on startup and startup is not defined'),
+        pytest.param(True, False, id='shutdown on connection loss not on startup but startup is defined as true'),
+    ]
+)
+class test_Consumer_WorkerShutdown(ConsumerTestCase):
+
+    def test_start_raises_connection_error(self,
+                                           broker_connection_retry_on_startup,
+                                           is_connection_loss_on_startup,
+                                           caplog, subtests):
+        c = self.get_consumer()
+        # in order to reproduce the actual behavior: if this is the startup, then restart count has not been
+        # incremented yet, and is therefore -1.
+        c.restart_count = -1 if is_connection_loss_on_startup else 1
+        c.app.conf['broker_connection_retry'] = False
+        c.app.conf['broker_connection_retry_on_startup'] = broker_connection_retry_on_startup
+        c.blueprint.start.side_effect = ConnectionError()
+
+        with subtests.test("Consumer raises WorkerShutdown on connection restart"):
+            with pytest.raises(WorkerShutdown):
+                c.start()
+
+        record = caplog.records[0]
+        with subtests.test("Critical error log message is outputted to the screen"):
```

Comment:
`outputted` -> `output`

Reply:
Seems like we can go [either way](https://pasttenses.com/output-past-tense).